src/bot.ts

```diff
@@ -0,0 +1,38 @@
+import fs from 'node:fs';
+import dotenv from 'dotenv';
+import { Client, Collection } from 'discord.js';
+import { Command } from './interfaces/Command';
+import { log } from './utils/log';
+dotenv.config();
+
+const client = new Client({ intents: ['GUILDS', 'GUILD_MESSAGES', 'GUILD_MESSAGE_REACTIONS'] });
+
+client.once('ready', () => {
+	log({
+		message: 'Bot is ready!',
+		section: 'BOT',
+		color: 'green',
+	});
+});
+
+const botCmds = new Collection();
+const cmdsFolder = fs.readdirSync('./src/commands');
+
+for (const file of cmdsFolder) {
+	const name = file.split('.')[0];
+	const cmd = import(`./commands/${name}.js`);
+
+	cmd.then(command => {
+		const cmdInstance: Command = new command.default();
+		botCmds.set(cmdInstance.commandStrucure.name, cmdInstance);
+	});
+}
+
+client.on('interactionCreate', i => {
+	if (!i.isCommand()) return;
+
+	const command = botCmds.get(i.commandName) as Command;
```

Comment:
If `botCmds` is typed, the type will be inferred. But we should use the non-null-assertion since the return type is `Command | undefined`.

```suggestion
	const command = botCmds.get(i.commandName)!;
```

(Maybe we should make a typeguard instead of an assertion)

Reply:
Fair, will fix in the next commit.